lib/response.js

```diff
@@ -491,6 +491,8 @@ function patch(Response) {
 
         if (self._charSet) {
             type = type + '; charset=' + self._charSet;
+        } else {
+            type = type + '; charset=utf-8';
```

Comment:
Choosing a default content type for a formatter should probably be the responsibility of the formatter, since binary content type should not have a charset.

Existing formatters are currently also tied to a particular charset since they set the `Content-Length` based on `Buffer.byteLength(data)` (Buffer.byteLength without a charset defaults to utf-8.) Example: https://github.com/restify/node-restify/blob/master/lib/formatters/json.js

(Setting Content-Length in the formatter should no longer be needed in recent NodeJS versions (>1.5.0), see: nodejs/node#1062.)

Perhaps it would make more sense to set the Content-Type header only after calling the formatter, providing the opportunity for the formatter to set a format appropriate default charset.

```js
        const results = formatter(self.req, self, body);
        
        if (self._charSet) {
            type = type + '; charset=' + self._charSet;
        }

        // Update header to the derived content type for our formatter
        self.setHeader('Content-Type', type);

        // Finally, invoke the formatter and flush the request with it's results
        return flush(self, results);
```

Reply:
I'm confused -  I thought the content type indirectly specifies the formatter, not the other way around?

Reply:
@DonutEspresso while the content type picks the formatter the formatter should probably pick the charset if it is unspecified.